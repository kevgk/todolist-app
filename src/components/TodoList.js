import React, { useReducer, useEffect } from 'react';
import { reducer } from '../store/reducer';
import TaskInput from './TaskInput';
import TaskCountFeedback from './TaskCountFeedback';
import TaskList from './TaskList';
import localStorageSaveJSON from '../utils/localStorageSaveJSON';
import { getTasksFromLocalStorage } from '../store/actions';
import useSkipFirstRender from '../utils/useSkipFirstRender';

export default function Todolist() {
	const [tasks, dispatch] = useReducer(reducer, []);

	useEffect(() => {
		getTasksFromLocalStorage(dispatch);
	}, [dispatch]);

	useSkipFirstRender(() => {
		localStorageSaveJSON('tasks', tasks);
	}, [tasks]);

	return (
		<div className='app'>
			<TaskCountFeedback tasks={tasks} />
			<h1>Todolist</h1>
			<TaskList tasks={tasks} dispatch={dispatch} />
			<TaskInput dispatch={dispatch} />
		</div>
	);
}
