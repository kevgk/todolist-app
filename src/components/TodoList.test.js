import React from 'react';
import { mount } from 'enzyme';

import TodoList from './TodoList';

it('renders correctly', () => {
	const wrapper = mount(<TodoList />);

	expect(wrapper).toMatchSnapshot();
});
