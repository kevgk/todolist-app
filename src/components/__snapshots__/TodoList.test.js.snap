// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="container"
  >
    <span
      aria-label="Todo count feedback"
      role="img"
    >
      ðŸ˜»
    </span>
  </div>
  <h1>
    Todolist
  </h1>
  <ul>
    <div
      className="noTasksHint"
    >
      Let's get productive!
    </div>
  </ul>
  <form
    className="taskContainer"
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="taskInput"
      placeholder="Add a task..."
    />
  </form>
</div>
`;
